packages/@aws-cdk/aws-secretsmanager/lib/secret.ts

```diff
@@ -127,9 +138,17 @@ export abstract class SecretBase extends cdk.Construct implements ISecret {
     }
   }
 
-  public toSecretString() {
-    this.secretString = this.secretString || new SecretString(this, 'SecretString', { secretId: this.secretArn });
-    return this.secretString;
+  public get secretString() {
```

Comment:
I would prefer this to be called “toSecretString” because it’s not a property of the secret, but rather a conversion method.

(Like toPipelineAction)

Reply:
> I would prefer this to be called “toSecretString” because it’s not a property of the secret

I disagree. It do see it as a property access (like `.secretArn`, but instead we access the VALUE of the secret). It just so happens that the value is of a complex type instead of a primitive (like `Date`).

It does not create a new resource, nor convert to anything that's used anywhere else in the construct tree. From this complex value object, you're still supposed to take `stringValue` or `jsonFieldValue`.

Reply:
OK, makes sense. Thanks for the explanation. What threw me off was the sugar methods `stringValue` and `jsonFieldValue`.